plugin/storage/cassandra/factory.go

```diff
@@ -157,7 +160,11 @@ func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger)
 
 // CreateSpanReader implements storage.Factory
 func (f *Factory) CreateSpanReader() (spanstore.Reader, error) {
-	return cSpanStore.NewSpanReader(f.primarySession, f.primaryMetricsFactory, f.logger, f.tracer.Tracer("cSpanStore.SpanReader"))
+	sr, err := cSpanStore.NewSpanReader(f.primarySession, f.primaryMetricsFactory, f.logger, f.tracer.Tracer("cSpanStore.SpanReader"))
+	if err != nil {
+		return sr, err
+	}
+	return spanstoremetrics.NewReaderDecorator(sr, f.metricsFactory), nil
```

Comment:
wrap CreateArchiveSpanReader to? This is where you can apply a label `name=primary|archive` before creating decorator

Reply:
@yurishkuro would we need to namespace the metric factory again? The factory doesn't expose a way to just label an existing namespace.

Reply:
why not - you can pass just the tags, with empty name

Reply:
oh okay got it - but the `name` label is already provided for v2 but not for v1 and we don't want to override it. Should we give it a different name? What about `type`?

Reply:
maybe `role` then

Reply:
`type` and `kind` are kind of the same.

Reply:
@yurishkuro Sounds good. One other thing to note about cassandra is that it currently holds two metrics factories that publish metrics under `jaeger_cassandra_***` and `jaeger_cassandra-archive_***` (https://github.com/jaegertracing/jaeger/blob/main/plugin/storage/cassandra/factory.go#L136-L137). Do we want to change these namespaces to match the same ones we're using for the decorator? So we could publish under `jaeger_storage_***` with the `role`, `kind`, and `name` (only for v2) labels.

Reply:
yes, I was thinking that too, since we're making a breaking change anyway let's make them more consistent.

Reply:
@yurishkuro Made this change and updated the PR with a summary of the breaking changes